src/core/api/commissioner_api.cpp

```diff
@@ -81,6 +81,16 @@ otError otCommissionerAddJoiner(otInstance *aInstance, const otExtAddress *aEui6
                                                            aTimeout);
 }
 
+otError otCommissionerGetNextJoinerInfo(otInstance *aInstance, uint16_t &aIterator, otJoinerInfo &aJoiner)
+{
+    otError error = OT_ERROR_NONE;
+
+    Instance &instance = *static_cast<Instance *>(aInstance);
+
+    error = instance.Get<MeshCoP::Commissioner>().GetNextJoinerInfo(aIterator, aJoiner);
```

Comment:
Suggestion: no need for `error` definition. We can just `return instance.Get<...`